src/main/java/org/openstreetmap/atlas/generator/sharding/PbfVerifier.java

```diff
@@ -0,0 +1,168 @@
+        {
+            final RawAtlasGenerator rawAtlasGenerator = new RawAtlasGenerator(pbfFile,
+                    AtlasLoadingOption.createOptionWithNoSlicing(), MultiPolygon.MAXIMUM);
+            try
+            {
+                rawAtlasGenerator.buildNoTrim();
+                final int currentCount = count.getAndIncrement();
+                if (currentCount % LOGGING_RATE == 0)
+                {
+                    logger.info("Processed " + currentCount + " PBF files.");
+                }
+            }
+            catch (final Exception e)
+            {
+                throw new CoreException("Error while building " + pbfFile.getName() + "!", e);
+            }
+        });
+        return 0;
+    }
+
+    public int checkForMissingPbfs(final HashMap<String, Rectangle> shardToBounds,
+            final List<String> pbfFileNames, final int expectedPbfCount)
+    {
+        // number of pbfs actually built
+        final int pbfFileCount = pbfFileNames.size();
+        final Integer missingPbfCount = expectedPbfCount - pbfFileCount;
+        if (missingPbfCount != 0)
+        {
+            logger.error("There are " + missingPbfCount + " pbfs missing!");
+            shardToBounds.keySet().forEach(pbfName ->
+            {
+                if (!pbfFileNames.contains(pbfName))
+                {
+                    logger.error(pbfName + " is missing!");
+                }
+            });
+            return 1;
+        }
+        logger.info("There are no pbfs missing!");
+        return 0;
+    }
+
+    @Override
+    protected int onRun(final CommandMap command)
+    {
+        final String pbfPath = (String) command.get(PBF_PATH);
+        final File slippyTileFile = (File) command.get(SLIPPY_TILE_FILE);
+        final Map<String, String> configuration = new HashMap<>();
+        configuration.put("fs.file.impl", "org.apache.hadoop.fs.LocalFileSystem");
+        final List<Resource> pbfFiles = FileSystemHelper.listResourcesRecursively(pbfPath,
+                configuration, new PbfFilePathFilter());
+        final List<String> pbfFileNames = pbfFiles.stream().map(file -> file.getName())
+                .collect(Collectors.toList());
+        final Time start = Time.now();
+        final HashMap<String, Rectangle> shardToBounds = parseSlippyTileFile(slippyTileFile);
+        final Integer expectedPbfCount;
+        try
+        {
+            // get the count of pbf tiles that should have been generated from the slippyTile file
+            expectedPbfCount = Integer.parseInt(slippyTileFile.firstLine());
+            // Since there can be empty PBF files, any missing file is a problem! Fail if there
+            // aren't the same # of pbfs as the slippytileFile expects, and log those that are
+            // missing
+            final int returnCode = checkForMissingPbfs(shardToBounds, pbfFileNames,
+                    expectedPbfCount);
+            if (returnCode != 0)
```

Comment:
Nit: more succinctly: `return (returnCode != 0) ? returnCode : buildAllPbfs(pbfFiles);`

Reply:
Actually there is still logging to do after this, we don't want to return the result of `buildAllPbfs()` just yet!

Reply:
Good call!